src/udev/udev-builtin-net_id.c

```diff
@@ -142,19 +139,16 @@ static int get_virtfn_info(sd_device *dev, struct netnames *names, struct virtfn
                         continue;
 
                 if (streq(syspath, virtfn_pci_syspath)) {
-                        if (!snprintf_ok(suffix, sizeof(suffix), "v%s", &dent->d_name[6]))
-                                return -ENOENT;
+                        if (asprintf(&suffix, "v%s", n) < 0)
+                                return -ENOMEM;
```

Comment:
for concatenating strings strjoin() is like a million times more efficient. (and morea readable, too...). 

hence:

```c
suffix = strjoin("v", n);
…
```